server/src/services/chat.service.ts

```diff
@@ -35,4 +42,91 @@ export class ChatService {
     });
     return messages;
   }
+
+  /**
+   * Create a private chat between two users and add them to it.
+   */
+  private async CreatePrivateChannel(user1: User, user2: User): Promise<void> {
+    await this.database.ExecuteInsideDatabaseHarness(async (connection) => {
+      await connection.query('CALL CreateChannel(?, ?);', [user1.id, user2.id]);
+    });
+  }
+
+  /**
+   * Figure out the private chatroom's ID for storage purposes.
+   */
+  private async GetChannelId(user1: User, user2: User): Promise<number> {
+    let result;
+    // Find the users' mutual channel that is not the global one.
+    await this.database.ExecuteInsideDatabaseHarness(async (connection) => {
+      result = await connection.query(
+        'SELECT u1.chatId FROM `' +
+          DatabaseTables.USER_CHAT +
+          '` AS u1 INNER JOIN userChat AS u2 ON u1.chatId = u2.chatId \
+          AND u1.userId = ? AND u2.userId = ? AND u1.chatId <> ?;',
+        [user1.id, user2.id, GLOBAL_CHAT_ID]
+      );
+    });
+    result = JSON.parse(JSON.stringify(result));
```

Comment:
Wow wow wow, back up here a second — if you want to copy an object use `const copy = { ...original }`.

The second thing is that I don’t really understand why would you want to copy and assign the same object… to itself? Am I missing something here? I presume it’s a result of some debugging process :P.

Reply:
I don't know, it wouldn't work without that 70th line :(

Reply:
Pushed a commit that resolves this issue.